spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala

```diff
@@ -68,57 +68,51 @@ object Predictor {
     })
   }
 
-  private[optim] def splitBatch[T: ClassTag](output: Activity, shareBuffer: Boolean, batchSize: Int)
+  private[optim] def splitTensor[T: ClassTag](output: Tensor[T],
+                                  shareBuffer: Boolean, batchSize: Int)
+    (implicit ev: TensorNumeric[T]): Array[Activity] = {
+    val result = if (shareBuffer) output else output.clone()
+    val size = result.size(1)
+    require(batchSize == size,
+      s"The batchSize is required to be $size, while actual is $batchSize")
+    val out = result.split(1)
+    out.asInstanceOf[Array[Activity]]
+  }
+
+  private[optim] def splitTable[T: ClassTag](output: Activity, shareBuffer: Boolean, batchSize: Int)
```

Comment:
The name is a little confusing. As you also split the tensor inside this function.

Reply:
Yes. As concluded,  the table represents the output structure and the split operation happened on tensor. When we find a tensor output, just use splitTensor to split it. I should remove splitTable here since there is no literally table split here and we just recursively probe into the deeper layers of the table until we find a tensor.

Reply:
Fix: remove spliTable and simply use splitBtach to do the recursive calls